Do not warn on nested route errors

ryardley opened the pull request:
Closes: #597 

### What are the changes and their implications?

* Do not warn on nested route errors
* Sibling Api folders are still compiled to pages/api like before
* Added missing unit test on duplicate route function
* Added unit test on entire page stage at the stream level
* Fixed bug in displaying code that was displaying too many matches.

### Checklist

- [x] Tests added for changes
- [x] User facing changes documented

### Breaking change: no

### Other information

<!-- IMPORTANT: Make sure to check the "Allow edits from maintainers" box below this window -->

ryardley commented:
There is contention here in #597 so we can’t merge this. Good to have eyes on it.

wKovacs64 commented:
Yep, I'm following that. Just fulfilling my review request in the event you convince Brandon. 😉

ryardley commented:
Ok adjusted the PR based on #597 to remove the Nested Route check. Please review and I will merge.

ryardley commented:
ooh @flybayer - you know what I am not sure maybe I will add a test case for that here.

ryardley commented:
Extra testing has been added. Found a bug where we were displaying all the routes multiple times depending on how many matches we were getting so I rewrote the duplication checking logic and added appropriate tests. Just checking all the windows stuff works then should be ready to merge.

merelinguist commented on packages/server/src/stages/pages/index.ts:
```diff
@@ -17,11 +78,14 @@ export const createStagePages: Stage = ({config, bus, getInputCache}) => {
+
     // Check for duplicate pages entries
-    const duplicatePages = getDuplicatePaths(entries, 'pages')
+    const duplicatePages = filterBy(duplicates, 'pages', 'api')
```
Nice refactoring!
